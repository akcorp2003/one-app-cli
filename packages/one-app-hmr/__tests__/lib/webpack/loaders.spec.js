/*
 * Copyright 2021 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

import {
  fileLoader,
  cssLoader,
  jsxLoader,
} from '../../../lib/webpack/loaders';

jest.mock('path', () => ({
  resolve: jest.fn(() => '/'),
}));

describe('fileLoader', () => {
  test('returns loader config for file types', () => {
    expect(fileLoader()).toMatchSnapshot();
  });
});

describe('cssLoader', () => {
  test('returns loader config for CSS files', () => {
    expect(cssLoader()).toMatchSnapshot();
  });
});

describe('jsxLoader', () => {
  test('returns loader config for JavaScript files', () => {
    expect(jsxLoader()).toEqual({
      exclude: /node_modules/,
      test: /\.jsx?$/,
      use: [
        {
          loader: require.resolve('babel-loader'),
          options: {
            babelrc: false,
            cacheDirectory: true,
            cwd: '/',
            plugins: [],
            presets: [
              [
                'amex',
                {
                  modern: true,
                  'preset-env': {
                    modules: false,
                  },
                },
              ],
            ],
          },
        },
      ],
    });
  });
});
