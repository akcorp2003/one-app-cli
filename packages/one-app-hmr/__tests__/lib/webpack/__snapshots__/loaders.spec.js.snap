// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cssLoader returns loader config for CSS files 1`] = `
Object {
  "exclude": /node_modules/,
  "test": /\\\\\\.\\(sa\\|sc\\|c\\)ss\\$/,
  "use": Array [
    Object {
      "loader": "style-loader",
    },
    Object {
      "loader": "css-loader",
      "options": Object {
        "importLoaders": 2,
        "modules": Object {
          "localIdentName": "[name]__[local]___[contenthash:base64:5]",
        },
      },
    },
    Object {
      "loader": "sass-loader",
    },
    Object {
      "loader": "postcss-loader",
      "options": Object {
        "postcssOptions": Object {
          "plugins": Object {
            "cssnano": Object {},
            "postcss-browser-reporter": Object {},
            "postcss-preset-env": Object {
              "browsers": "last 2 versions",
            },
          },
          "syntax": "postcss-scss",
        },
        "sourceMap": true,
      },
    },
  ],
}
`;

exports[`fileLoader returns loader config for file types 1`] = `
Object {
  "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\|svg\\|png\\|jpg\\|jpeg\\|gif\\|webm\\)\\(\\\\\\?\\.\\*\\)\\?\\$/,
  "use": Array [
    Object {
      "loader": "file-loader",
      "options": Object {
        "name": "../assets/[name].[ext]",
      },
    },
  ],
}
`;

exports[`jsxLoader returns loader config for JavaScript files 1`] = `
Object {
  "exclude": /node_modules/,
  "test": /\\\\\\.jsx\\?\\$/,
  "use": Array [
    Object {
      "loader": "/Users/nkiboi/one/oneamex/github/one-app-cli/node_modules/babel-loader/lib/index.js",
      "options": Object {
        "babelrc": false,
        "cacheDirectory": true,
        "cwd": "/Users/nkiboi/one/oneamex/github/one-app-cli/packages/one-app-hmr/lib",
        "plugins": Array [],
        "presets": Array [
          Array [
            "amex",
            Object {
              "modern": true,
              "preset-env": Object {
                "modules": false,
              },
            },
          ],
        ],
      },
    },
  ],
}
`;
